tests/static-analysis/reduce.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+include __DIR__ . '/../../vendor/autoload.php';
+
+use loophp\collection\Collection;
+use loophp\collection\Contract\Collection as CollectionInterface;
+
+$sum = static fn (int $carry, int $value): int => $carry + $value;
+$concat = static fn (string $carry, string $string): string => sprintf('%s%s', $carry, $string);
+
+/**
+ * @param CollectionInterface<int, int> $collection
+ */
+function reduce_checkListInt(CollectionInterface $collection): void
+{
+}
+
+/**
+ * @param CollectionInterface<int, string> $collection
+ */
+function reduce_checkListString(CollectionInterface $collection): void
+{
+}
```

Comment:
rather than two list checks we should have a list and a "map" check, e.g. this should accept `CollectionInterface<string, string>`

Reply:
Done.